modules/library/main/src/main/java/org/geotools/data/collection/CollectionDataStore.java

```diff
@@ -37,8 +37,10 @@
 
 import com.vividsolutions.jts.geom.Geometry;
 
-
+@Deprecated
 /**
+ * AbstractDataStore is deprecated. Migrate to ContentDataStore.
```

Comment:
In this case we will migrate to a new implementation. You can see I made a small "in-line" collection data store inside DataUtilities. Maybe we can take that implementation and replace the contents of this class.